### PR TITLE
pythonPackages.stestr: init 3.2.0

### PR DESCRIPTION
**Pull request by superherointj**
* pythonPackages.stestr: init 3.2.0
* git-machete: fixes tests

Tests are disabled until `openstack-hacking` is merged (I expect to do so eventually).

I have fixed `git-manchete` to show this works.

### CODE REVIEW

**Review by blitz — APPROVED** [review]
The git-machete changes look good to me.

File: pkgs/development/python-modules/stestr/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    pbr
+    cliff
+    subunit
+    fixtures
+    testtools
+    pyyaml
+    voluptuous
+  ];
+
+  # Tests should only run on passthru to avoid infinite recursion from `openstack-hacking`.
```
Comment (risicle) [inline pkgs/development/python-modules/stestr/default.nix:25]: Other ways people sometimes achieve this is to use an overridden version of one of the packages with `doCheck` disabled _just for use in the tests_. The main version seen by the end user of course does have the tests enabled.

File: pkgs/development/python-modules/stestr/tests.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    ddt
+    doc8
+    # openstack-hacking # To be enabled after `openstack-hacking` is merged.
+    sphinx
+  ];
+
+  # checkPhase = ''
+  #   stestr
+  # '';
+}
```
Comment (risicle) [inline pkgs/development/python-modules/stestr/tests.nix:27]: Admirable that you've done this, but until the other package has been merged, this is probably better left out. You'd need another PR or commit to come back and enable these tests anyway. You could just add them then.

Comment (superherointj) [reply pkgs/development/python-modules/stestr/tests.nix:27]: Enabling is less work than typing it all. It acts as a placeholder. I'm trying to port OpenStack and currently I'm upstreaming parts. As I get parts in place I will be enabling tests. The other option I had was to make a huge PR. But there was also a huge PR that failed. I prefer leaving these comments parts. It has just proved useful, with your help of course.

Comment (risicle) [reply pkgs/development/python-modules/stestr/tests.nix:27]: Have you considered tracking your OpenStack work as a project? https://github.com/NixOS/nixpkgs/projects

**Comment by superherointj** [comment]
Can this be merged?
I need this for the OpenStack PR. I have already 25 openstack packages working that I want to upstream.

**Comment by mweinelt** [comment]
Any reason you don*t include the openstack-hacking package in this pr, so we can work on the infinite recursion here?

**Comment by superherointj** [comment]
> Any reason you don't include the openstack-hacking package in this pr, so we can work on the infinite recursion here?

In a few packages tests work properly. But as tests are enabled in more packages it is going to loop. Even using (openstack-hacking.overridePythonAttrs (oldAttrs: rec { doCheck = false; })) and `passthru.tests`. I have a lot of working tests already. Do you want me to push them as well? But as I said, as we add more of them, and workaround things, it is still going to break, at least this was my experience. It gets ugly. :)

**Comment by superherointj** [comment]
I will merge this PR with OpenStack.

**Pull request closed without merging**